### PR TITLE
BPTL Dashboard: Add API endpoint to generate kit data csv file

### PR DESCRIPTION
**Pull request by abhinavjonnada82**
Title^^

This PR addresses following issue: https://github.com/episphere/connect/issues/816

Related PR: https://github.com/episphere/biospecimen/pull/630

### CODE REVIEW

**Review by jhflorey — APPROVED** [review]

**Review by ramavavi — APPROVED** [review]

**Comment by ramavavi** [comment]
Looks good to me.

File: utils/biospecimen.js [hunk 1]
```diff
@@ -752,6 +752,23 @@ const biospecimenAPIs = async (req, res) => {
...
+        }
+        try {
+            const queryReceivedDateTimestamp = req.query.receivedDateTimestamp;
+            if(Object.keys(queryReceivedDateTimestamp).length === 0) return res.status(404).json(getResponseJSON('Please include parameter to filter data.', 400));
+            const { queryKitsByReceivedDate } = require('./firestore');
+            const response = await queryKitsByReceivedDate(queryReceivedDateTimestamp);
+            return res.status(200).json({data: response, code:200});
+        }
+        catch(error) {
+            console.error(error);
```
Comment (we-ai) [inline utils/biospecimen.js:767]: A bit more context about the printed error.

File: utils/biospecimen.js [hunk 2]
```diff
@@ -752,6 +752,23 @@ const biospecimenAPIs = async (req, res) => {
         }
     }
 
+    else if(api === 'getKitsByReceivedDate') {
+        if(req.method !== 'GET') {
+            return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
+        }
+        try {
+            const queryReceivedDateTimestamp = req.query.receivedDateTimestamp;
```
Comment (we-ai) [inline utils/biospecimen.js:760]: Could be undefined.

File: utils/biospecimen.js [hunk 3]
```diff
@@ -752,6 +752,23 @@ const biospecimenAPIs = async (req, res) => {
...
+            if(Object.keys(queryReceivedDateTimestamp).length === 0) return res.status(404).json(getResponseJSON('Please include parameter to filter data.', 400));
```
Comment (we-ai) [inline utils/biospecimen.js:761]:
```suggestion
            if (queryReceivedDateTimestamp.length === 0) return res.status(404).json(getResponseJSON('Please include parameter to filter data.', 400));
```

**Pull request merged**